src/js/cypress/Browser.ts

```diff
@@ -168,74 +135,159 @@ export class Browser {
     }
 
-    getElement(selector: string) {
-        return cy.get(selector);
+    getElement(selector: string, timeout: BrowserTimeout = "sm") {
+        return this.get(selector, { timeout: this.getTimeoutTime(timeout) });
     }
 
     uploadFileFromFilePath(selector: string, filePath: string) {
-        return cy.get(selector).selectFile(filePath, { force: true });
+        return this.get(selector).selectFile(filePath, { force: true });
     }
 
     getElementText(selector: string) {
-        return cy.get(selector).invoke("text");
+        return this.get(selector).invoke("text");
+    }
+
+    getEachElementTexts(selector: string) {
+        const results: string[] = [];
+
+        this.get(selector).each((item) => {
+            results.push(item.text());
+        });
+
+        return cy.wrap(results);
     }
 
     dispatchEvent(selector: string, event: Event) {
-        return cy.get(selector).then(($el) => {
+        return this.get(selector).then(($el) => {
             // setting the value onto element and dispatching input
             // event should trigger React's change event
             $el[0].dispatchEvent(event);
         });
     }
 
     check(selector: string) {
-        cy.get(selector).check();
+        this.get(selector).check();
     }
 
     uncheck(selector: string) {
-        cy.get(selector).uncheck();
+        this.get(selector).uncheck();
     }
 
     getAttribute(selector: string, attribute: string) {
-        return cy.get(selector).invoke("attr", attribute);
+        return this.get(selector).invoke("attr", attribute);
+    }
+
+    getCssProperty(selector: string, property: string) {
+        return this.get(selector)
+            .invoke("css", property)
+            .then(({ value }) => value);
+    }
+
+    retry(
+        cb: () => Cypress.Chainable<boolean>,
+        become?: boolean,
+        delay?: BrowserTimeout,
+        timeout?: BrowserTimeout,
+    ): void;
+
+    retry<T>(
+        cb: () => Cypress.Chainable<T>,
+        become: T,
+        delay?: BrowserTimeout,
+        timeout?: BrowserTimeout,
+    ): void;
+
+    retry<T = boolean>(
+        cb: () => Cypress.Chainable<T>,
+        become?: T,
+        delay: BrowserTimeout = "zero",
+        timeout: BrowserTimeout = "md",
+    ) {
+        cy.until({
+            it: cb,
+            become,
+            delay: defaultSettings.timeouts[delay],
+            timeout: defaultSettings.timeouts[timeout],
+        });
+    }
+}
+
+export class IframeBrowser extends Browser {
+    #document: Cypress.Chainable;
+
+    get(selector: GetParams[0], options?: GetParams[1]) {
+        return this.#document.find(selector, options);
+    }
+
+    xpath(selector: XpathParams[0], params?: XpathParams[1]) {
+        return this.#document.xpath(selector, params);
+    }
+
+    constructor(selector: string, settings: BrowserSettings, iframeTimeout: BrowserTimeout) {
+        super(settings);
+
+        this.#document = cy
```

Comment:
We can just redefine this.document to be the iframe, use `this.document.get()` in the definition of `get` in the parent class and avoid having to redefine other functions